src/shared/codelens/localLambdaRunner.ts

```diff
@@ -0,0 +1,329 @@
+}
+
+export interface SAMTemplateEnvironmentVariables {
+  [resource: string]: {
+      [key: string]: string
+  }
+}
+
+export class LocalLambdaRunner {
+
+  private static readonly TEMPLATE_RESOURCE_NAME: string = 'awsToolkitSamLocalResource'
+  private static readonly SAM_LOCAL_PORT_CHECK_RETRY_INTERVAL_MILLIS: number = 125
+  private static readonly SAM_LOCAL_PORT_CHECK_RETRY_TIMEOUT_MILLIS_DEFAULT: number = 30000
+
+  private _baseBuildFolder?: string
+  private readonly _debugPort?: number
+
+  public constructor(
+      private readonly configuration: SettingsConfiguration,
+      private readonly localInvokeArgs: LambdaLocalInvokeArguments,
+      debugPort: number | undefined,
+      private readonly runtime: string,
+      private readonly outputChannel: vscode.OutputChannel,
+      private readonly processInvoker: SamCliProcessInvoker,
+      private readonly taskInvoker: SamCliTaskInvoker,
+      private readonly debugConfig: vscode.DebugConfiguration,
+      private readonly codeRootDirectoryPath: string,
+      private readonly onWillAttachDebugger?: () => Promise<void>,
+      private readonly onDidSamBuild?: () => Promise<void>
+  ) {
+      if (localInvokeArgs.debug && !debugPort) {
+          throw new Error('Debug port must be provided when launching in debug mode')
+      }
+
+      this._debugPort = debugPort
+  }
+
+  public async run(): Promise<void> {
+      try {
+          logger.info(
+              localize(
+                  'AWS.output.sam.local.start',
+                  'Preparing to run {0} locally...',
+                  this.localInvokeArgs.handlerName
+              )
+          )
+
+          const inputTemplate: string = await this.generateInputTemplate(this.codeRootDirectoryPath)
+          const samBuildTemplate: string = await this.executeSamBuild(this.codeRootDirectoryPath, inputTemplate)
+
+          await this.invokeLambdaFunction(samBuildTemplate)
+
+      } catch (err) {
+          // TODO: logger.error?
+          console.log(err)
+          const error = err as Error
+
+        // TODO: Define standard/strategy. Sometimes err.message is/isn't part of msg "Error: {0}"
+          this.outputChannel.appendLine(
+              localize(
+                  'AWS.output.sam.local.error',
+                  'Error: {0}',
+                  error.message
+              )
+          )
+
+          vscode.window.showErrorMessage(
+              localize(
+                  'AWS.error.during.sam.local',
+                  'An error occurred trying to run SAM Application locally: {0}',
+                  error.message
+              )
+          )
+
+          return
+      }
+
+  }
+
+  public get debugPort(): number {
+      if (!this._debugPort) {
+          throw new Error('Debug port was expected but is undefined')
+      }
+
+      return this._debugPort
+  }
+
+  private async getBaseBuildFolder(): Promise<string> {
+      if (!this._baseBuildFolder) {
+          const baseBuildDir = path.join(
```

Comment:
Commit eddc6f0 broke this. Since the code is moving to a new file, it is ideal to fix it now.
The [change](https://github.com/aws/aws-toolkit-vscode/commit/eddc6f00b74c6e682804bd7757b2a084a173f777#diff-ba4a0343c78ed5eb2f12113bb6521508L263) replaced `mkdtempAsync` with `mkdir`. This causes errors illustrated below when trying to run any function locally, and restarting the toolkit is the only fix.

![image](https://user-images.githubusercontent.com/39839589/53905099-865c1180-3ffc-11e9-81ae-2f8c31e084ba.png)

TODO
- [ ] Revert back to using `mkdtempAsync`
- [ ] ...and the call should be against `build-`